src/mlpack/CMakeLists.txt

```diff
@@ -158,3 +159,80 @@ endif ()
 # If we are building Markdown documentation, we have to run some setup after we
 # recurse into methods/.  If not, this function is empty.
 post_markdown_setup()
+
+# Installing CMake Helper
+
+# Joins arguments and places the results in ${result_var}.
+function(join result_var)
+set(result )
+foreach (arg ${ARGN})
+  set(result "${result}${arg}")
+endforeach ()
+set(${result_var} "${result}" PARENT_SCOPE)
+endfunction()
+
+# Sets a cache variable with a docstring joined from multiple arguments:
+#   set(<variable> <value>... CACHE <type> <docstring>...)
+# This allows splitting a long docstring for readability.
+function(set_verbose)
+cmake_parse_arguments(SET_VERBOSE "" "" "CACHE" ${ARGN})
+list(GET SET_VERBOSE_CACHE 0 type)
+list(REMOVE_AT SET_VERBOSE_CACHE 0)
+join(doc ${SET_VERBOSE_CACHE})
+set(${SET_VERBOSE_UNPARSED_ARGUMENTS} CACHE ${type} ${doc})
+endfunction()
+
+include(CMakePackageConfigHelpers)
+
+set_verbose(MLPACK_CMAKE_DIR ${CMAKE_INSTALL_LIBDIR}/cmake/mlpack CACHE STRING
+          "Installation directory for cmake files, relative to "
+          "${CMAKE_INSTALL_PREFIX}.")
+set(version_config ${PROJECT_BINARY_DIR}/mlpack-config-version.cmake)
+set(project_config ${PROJECT_BINARY_DIR}/mlpack-config.cmake)
+set(targets_export_name mlpack-targets)
```

Comment:
Just for consistency, do you mind making the CMake variables all-caps?  All of the other CMake variables (or, well, most of them) are like that. :+1:

Reply:
Yes, I'll do that :-)